nextjs-wordles/pages/index.js

```diff
@@ -1,17 +1,151 @@
+
+
+
+let DEFAULT_STATE = Array.from({length: ROW_COUNT}, () => Array.from({length: COL_COUNT}, () => ({value:"", state:""})))
+//let DEFAULT_STATE = Array.from((Array(ROW_COUNT)), () => Array.from((Array(COL_COUNT), () => FINAL_VALUE_1))
 
 export default function Home() {
+  
+  let [board, setBoardState] = useState(DEFAULT_STATE );
+
+  const [keyboardKey, setKeyboardKey] = useState('');
+  const [currentRow, setCurrentRow] = useState(0);
+  const [currentObject, setCurrentObject] = useState(0);
+  
+  const isSpecialKey = key => SPECIAL_KEYS.includes(key);
+  function isAllowedLetter(letter) {
+    return ALLOWED_LETTERS.includes(letter)
+  }
+//======================================================
+// Funkcja handleKeyPress - pozwala keyboardKey odbierać wartości z klawiatury fizycznej
+//======================================================
+useEffect(() => {
+  const handleKeyPress = (event) => {
+    const letter = event.key
+    if(isAllowedLetter(letter)) {
+      console.log("Naciśnięto klawisz: " + letter);
+     setKeyboardKey(letter.toUpperCase())
+    }else {
+      console.log("Nie ma")
+    } 
+  }
+  document.addEventListener("keydown", handleKeyPress)
+  setKeyboardKey("")
+})
+
+//======================================================
+// Aktualizowanie stanu tablicy
+//======================================================
+
+  useEffect(() => {
+    if(keyboardKey === '') return 
+
+    if(!isSpecialKey(keyboardKey)) {
+      updateBoard(keyboardKey);
+    }else  if(keyboardKey === "Delete" ||keyboardKey ===  "DELETE" || keyboardKey ===  "BACKSPACE") {
+      deleteLetter()
+    }else if (keyboardKey === 'Enter' ||keyboardKey ===  "ENTER") {
+      verifyState();
+    }
+  },[keyboardKey])
+
+  const updateBoard = (key) => {
+
+    let updatedBoard = [...board];
+    let uptatedCurrentObject = currentObject
+
+    if(uptatedCurrentObject >= 5) return uptatedCurrentObject = 5
+    if (board[currentRow][4].value === "") {
+      updatedBoard[currentRow][uptatedCurrentObject] = {value: key, state:""}
+    }
+
+    setBoardState(updatedBoard);
+    setCurrentObject(uptatedCurrentObject + 1)
+  }
+
+  function verifyState() {
+    if(isWordCorrect()) {
+       compare()
+       nextGameWin()
+    }
+    if(!isWordCorrect() && currentRow === 5){
+      compare()
+      nextGameLose()
+    }else{
+      compare()
+        }
+
+    setCurrentRow(currentRow + 1)
+    setCurrentObject(0)
+  }
+//======================================================
+// Sprawdzenie checkWord
+//======================================================
+  const deleteLetter = () =>{
+    let updatedBoard = [...board];
+    let objcetToDelete = currentObject -1
+    if(objcetToDelete < 0) return objcetToDelete = 0
+    updatedBoard[currentRow][objcetToDelete].value = "";
+
+    setBoardState(updatedBoard);
+    setCurrentObject(objcetToDelete)
+    setKeyboardKey("")
```

Comment:
Tu jest ten case z użytkownikiem klikającym dwa razy literę O w "BOOK". Żeby nie używać wstawiania nowego pustego stringu w celu wychwycenia zmiany i przerenderowania tablicy, można użyć obiektu.

Reply:
Tutaj bym potrzebował pomocy nie za bardzo widzę jak można to zrobić za pomocą obiektu

Reply:
Rozumiem. 
Może counter pisania?

```
 let [count, setKeyboardKey] = useState({key:'', lettersTypes: 0});

setKeyboardKey({key: key, lettersTyped: count++})

```

Reply:
@AndrzejGrabarz  popchnijmy ten PR bez tej zmiany i zerkniemy na to potem przy okazji weryfikacji dodawania słownika do sprawdzania istniejących słów.